pypicloud/cache/dynamo.py

```diff
@@ -22,6 +25,78 @@
 LOG = logging.getLogger(__name__)
 
 
+def _path_depth(package):
+    """Count the slashes in path, or return infinity on no path."""
+    path = package.data.get('path')
+    if not path:
+        return float('+inf')
+    return path.count('/')
+
+
+def _decide_between_versions(contender, current):
+    """Decide between packages with the same filename on different paths."""
+
+    if contender.last_modified < current.last_modified:
+        return contender
```

Comment:
Just realized I can't make a failing test for this.

With a list of packages and a select element from that list, regardless of the the order of the list and the element we select, we're always going to converge on the earliest element. Even if the only ordering we have is partial (as is our case, we were only able to tell what's earlier, not equal).

However, not everybody has poset theory present when reading the code (neither did I, until I exhausted all the possible test formulations). Leaving the redundant check for clarity and to make it reusable in other contexts without gotchas.